notion_client/client.py

```diff
@@ -61,8 +62,17 @@ def _build_request(self, method, path, body):
         return self.client.build_request(method, path, json=body)
 
     def request(self, path, method, query=None, body=None, auth=None):
-        request = self._build_request(method, path, body)
-        return self.client.send(request)
+        try:
+            request = self._build_request(method, path, body)
+            response = self.client.send(request)
+            response.raise_for_status()
+            return response
```

Comment:
Is it possible that we handle correctly an error that is raised by any other line than `response.raise_for_status()`? If not, we should rather have these other lines outside the try-except to let real errors (i.e. not related to the HTTP status code) being raised directly.

Reply:
I tested that If not `raise_for_status()`, HTTPStatusError is not occured and 400 / 500 errors will be ignored so we can't parse Notion's specified known errors. 
If not implemented, checking 400 / 500 errors logic must be specified on our code.
So rather than this way, I'd rather use httpx's parsing 400 / 500 logic.
400 / 500 errors that notion not specified will be filtered on here:
```py
def parse_api_error_response_body(body: dict) -> Union[APIErrorResponseBody, None]:
    if not is_api_error_code(body["code"]):
        return None
```
on errors.py line 94

Reply:
Yeah I'm good with `raise_for_status()`! I'm just saying we could have less lines of code in the try block. :)